### PR TITLE
Hey what if I made Sleeping Carp better at nonlethal takedowns and also deflect with combat mode instead of throw mode (but cost more)

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

It's been a hot minute hasn't it?

When I initially reworked Sleeping Carp, we didn't have combat mode. Now that we do, and that Sleeping Carp has substantially less defensive power to justify having to make a choice between deflection and attacking, it's probably about time we updated this aspect back to what it was before my rework. Sorta.

Now, we can have all the deniability of the previous method, while also letting you reliably protect yourself from ranged attacks at all times while it matters. Because of this, I increased the price up to 17 TC because of this change just to be on the safe side. The higher uptime of projectile immunity while also being able to attack during that time makes this a lot stronger overall.

Secondly, Sleeping Carp presently just isn't as good as a good ol' baton. It takes a lot more hits to accomplish the same task that a baton can. Many people feel like they can't even reasonably fight anyone for fear of the baton, or they would rather use a baton and kill someone at their leisure. So we've updated some of the moves in order to facilitate Sleeping Carp as a substantial contender for 1v1 fighting, and lessen the need for a baton by adding a lot more Stamina damage overall to the various attacks;

**Keelhaul**: Now a Shove Shove combo. Does literally zero lethal damage, but now temporarily blinds and dizzies the target as well as its previous effects. The amount of lethal damage it did was...extremely small, so this isn't a particularly big loss.

**Grabs and Shoves**: Deal some amount of stamina damage (20). You need to be in combat mode in order to perform these special attacks (more deniability). Grabbing someone while they have 80 Stamina damage or more will cause them to fall unconscious. Yes, I really did just want to add a Vulcan Nerve Pinch, what do you want from me?

That's it actually. Oh, I guess they are heavy sleepers now too. Because its funny.

## Why It's Good For The Game

I often get told (read: thrown various insults and slurs at me while mentioning this as the justification) that Sleeping Carp is not very strong anymore since it lost all that invisible armor I added way back + I removed the stuns in my initial rework. This made some people upset (I think at least one person wished for my death).

So, having given it at least 2 years, I wanted to recapture parts of what made the older Sleeping Carp (before my rework) strong, some of the benefits of the new version, and introduce a brand new aspect; nonlethal takedowns. This makes it beneficial for pacifists, as well as for kidnapping.

This should not meaningfully make Sleeping Carp any stronger against the things that typically ruin its day. I suspect in a straight joust with a baton, Sleeping Carp will still struggle. But against what should be its strong points (lone targets and ranged weapons), it will be strong once again rather than clumsily unable to do very much at all. 

## Changelog
:cl:
balance: Harnessing Shoreline Quay (bluespace energy, probably), a mystical energy (total bullshit) that permeates the Astral Waterways (bluespace quantum dimensions, probably), Sleeping Carp users can now once against deflect projectiles with their bare hands when focused in on battle (in combat mode). 
balance: The Keelhaul technique is now nonlethal (a philosophical acknowledgement of the familial bond of sleep and death), but causes the target to become temporarily blind and dizzy along with its previous effects.
balance: Sleeping carp users, while in combat mode, deal Stamina damage with their grabs and shoves. If the target of their grab has enough Stamina damage (80), they are knocked unconscious from a well placed nerve pinch.
balance: Sleeping carp users find it very hard to wake up once they fall asleep....
/:cl:

### CODE REVIEW

**Comment by Borisvanmemes** [comment]
I am going to kidnap so many people with this.

**Comment by YuliceTheDruid** [comment]
This will make North Star SCarp so fucking strong, the increased cost will make that harder to pull off but if someone pulls off the TC cost or gets it on sale they are going to be a GOD

**Comment by necromanceranne** [comment]
It was always pretty strong, I'm confident the price increase will probably keep that from happening too readily.

**Comment by Likteer** [comment]
>  a mystical energy that permeates the Astral Waterways (bluespace quantum dimensions)

Just don't. Blue Space stands for BullShit.

Except clown car, His Grace and jokes, traitor items cap at 15TC.
I'd go with that so one could take some cheap utility,
or didn't have to save as much for something better.

<details>
<summary>I didn't read that other comment before posting</summary>
I didn't think about the combo.
My intentions are pure.
</details>

**Comment by SimplySmiley123** [comment]
> It was always pretty strong, I'm confident the price increase will probably keep that from happening too readily.

Sleeping Carp's only really strong when you got some way to incapacitate someone before using it (read: stun combat my beloathed) or somehow gaining the ability to pull off a combo _before_ your target notices and pulls out a stun baton they likely yoinked from a killed Beepsky.

**Comment by mc-oofert** [comment]
damn well will people still use easy nonlethal takedowns to follow up by murdering them while they cant do shit

**Comment by DaydreamIQ** [comment]
I don't really see the point of adding more downsides in the form of heavy sleeper considering it already makes you unable to use ranged weapons (Plus thematically it makes ZERO sense). Plus changing it from throwmode to combat mode is basically a soft revert to OLD Scarp when it was a passive effect, the main reason I believe throwmode was used is because forced you to pick between being offensive with it off or defensive with it on.

**Comment by Jacquerel** [comment]
> I don't really see the point of adding more downsides in the form of heavy sleeper considering it already makes you unable to use ranged weapons (Plus thematically it makes ZERO sense). Plus changing it from throwmode to combat mode is basically a soft revert to OLD Scarp when it was a passive effect, the main reason I believe throwmode was used is because forced you to pick between being offensive with it off or defensive with it on.

You are replying to the PR made by the person who made it deflect during throw mode. 
You do not need to speculate or explain to her what the reasoning for the particulars of that change were because she was the one who made it and knows them already.

**Comment by DaydreamIQ** [comment]
> 
> You are replying to the PR made by the person who made it deflect during throw mode. You do not need to speculate or explain to her what the reasoning for the particulars of that change were because she was the one who made it and knows them already.

Okay but the point still stands. Why wouldn't you always have combat mode in a fight using this? Surely it'd make more sense to have the effect of the shoves and the deflect something that you have to trade one for the other.

File: code/datums/martial/sleeping_carp.dm [hunk 1]
```diff
@@ -66,56 +66,80 @@
...
+	var/grab_log_description = "grabbed"
+	attacker.do_attack_animation(defender, ATTACK_EFFECT_PUNCH)
+	playsound(get_turf(defender), 'sound/weapons/punch1.ogg', 25, TRUE, -1)
+	if(defender.stat != DEAD && !defender.IsUnconscious() && defender.getStaminaLoss() >= 80) //We put our target to sleep.
+		defender.visible_message(
+			span_danger("[attacker] carefully pinch a nerve in [defender]'s neck, knocking them out cold"),
+			span_userdanger("[attacker] pinches something in your neck, and you fall unconscious!"),
+		)
+		grab_log_description = "grabbed and nerve pinched"
+		defender.apply_effect(10 SECONDS, EFFECT_UNCONSCIOUS)
```
Comment (MrMelbert) [inline code/datums/martial/sleeping_carp.dm:103]: Just use `Unconscious` if you aren't passing in an armor value to `apply_effect`
```suggestion
		defender.Unconscious(10 SECONDS)
```

File: code/datums/martial/sleeping_carp.dm [hunk 2]
```diff
@@ -66,56 +66,80 @@
...
 /datum/martial_art/the_sleeping_carp/disarm_act(mob/living/attacker, mob/living/defender)
+	if(!can_deflect(attacker)) //allows for deniability
+		return ..()
+
 	add_to_streak("D", defender)
 	if(check_streak(attacker, defender))
 		return TRUE
+
+	attacker.do_attack_animation(defender, ATTACK_EFFECT_PUNCH)
+	playsound(get_turf(defender), 'sound/weapons/punch1.ogg', 25, TRUE, -1)
```
Comment (MrMelbert) [inline code/datums/martial/sleeping_carp.dm:135]:
```suggestion
	playsound(defender, 'sound/weapons/punch1.ogg', 25, TRUE, -1)
```

File: code/datums/martial/sleeping_carp.dm [hunk 3]
```diff
@@ -13,13 +13,13 @@
 	. = ..()
 	if(!.)
 		return
-	target.add_traits(list(TRAIT_NOGUNS, TRAIT_HARDLY_WOUNDED, TRAIT_NODISMEMBER), SLEEPING_CARP_TRAIT)
+	target.add_traits(list(TRAIT_NOGUNS, TRAIT_HARDLY_WOUNDED, TRAIT_NODISMEMBER, TRAIT_HEAVY_SLEEPER), SLEEPING_CARP_TRAIT)
```
Comment (MrMelbert) [inline code/datums/martial/sleeping_carp.dm:16]: Should make this list of traits a list on the datum itself, that we pass to `add` and `remove`. Makes it easier to modify going forward

**Comment by HoodMilk** [comment]
> I don't really see the point of adding more downsides in the form of heavy sleeper considering it already makes you unable to use ranged weapons (Plus thematically it makes ZERO sense). Plus changing it from throwmode to combat mode is basically a soft revert to OLD Scarp when it was a passive effect, the main reason I believe throwmode was used is because forced you to pick between being offensive with it off or defensive with it on.

It gives you heavy sleeper because it's SLEEPING carp :)

**Comment by necromanceranne** [comment]
> Okay but the point still stands. Why wouldn't you always have combat mode in a fight using this? Surely it'd make more sense to have the effect of the shoves and the deflect something that you have to trade one for the other.

I'll save you the need to speculate. I did partially already explain the reasoning for the change, but I'll be more specific and include the historical changes.

My initial rework of Sleeping Carp came with considerable defensive buffs. You were a real bastard to actually put down, and could take a substantial amount of damage from most all sources. This was an effect that took place regardless of whether or not you could deflect. Not wanting to make it nearly impossible to meaningfully affect the Sleeping Carp user, the user had to make the conscious choice of being bulletproof, at the expense of being unable to fight. Throw mode made sense for that. It was so that, even though they were still very hard to fight, they still had some limitations. The change was a limitation, through and through, that made sense only with that design.

Then they no longer had that design, but still had the limitation. That genuinely doesn't seem logical to me anymore, as it deprives Sleeping Carp of one of its major upsides. Safety from projectile weapons. 

Yes, most of the time, you're going to be in combat mode when fighting. That's the point. I intend for that to be the case. There doesn't need to be, during a fight, any thought needed to being able to deflect ranged weapons because there is no real meaningful gameplay attached to having to micromanage that which isn't really annoying.

**Comment by necromanceranne** [comment]
> It gives you heavy sleeper because it's SLEEPING carp :)

Putting the Sleep in Sleeping Carp (just an eepy guy)

File: code/datums/martial/sleeping_carp.dm (hunk 3 above)
Comment (necromanceranne) [reply code/datums/martial/sleeping_carp.dm:16]: sure

**Comment by necromanceranne** [comment]
> Just don't. Blue Space stands for BullShit.

For your benefit, I've made it more obvious that this is **A Joke** and it is all bullshit.
 
> Except clown car, His Grace and jokes, traitor items cap at 15TC. I'd go with that so one could take some cheap utility, or didn't have to save as much for something better.

Holoparasites are 18 TC, and Double-Energy Swords are 16 TC, and the Elite modsuit is 16 TC. A lot of the reasoning as far as I can tell for the price point being this high is because they are either just very old, or they're presumed to be doing double duty in some way, such as providing any combination of offensive, defensive or utilitarian benefits. But there is no consistent design behind this, given the large price variance.

**Comment by ATH1909** [comment]
> > a mystical energy that permeates the Astral Waterways (bluespace quantum dimensions)
> 
> Just don't. Blue Space stands for BullShit.
> 
> Except clown car, His Grace and jokes, traitor items cap at 15TC. I'd go with that so one could take some cheap utility, or didn't have to save as much for something better.
> 
> I didn't read that other comment before posting

holoparasites, hello??

**Comment by Likteer** [comment]
> _addresses everything in a chill manner_ 

@necromanceranne Thank you

**Comment by Overuseddart** [comment]
I feel like while heavy sleeper is a funny joke, you have to keep in mind the quirk allows stuff like infinite table slams.

**Comment by necromanceranne** [comment]
Uh, how? Heavy Sleeper only affects sleep.

**Comment by Overuseddart** [comment]
> Uh, how? Heavy Sleeper only affects sleep.

Oh I thought it increased stun times sorry my bad

**Comment by Redrover1760** [comment]
> I don't really see the point of adding more downsides in the form of heavy sleeper considering it already makes you unable to use ranged weapons (Plus thematically it makes ZERO sense). Plus changing it from throwmode to combat mode is basically a soft revert to OLD Scarp when it was a passive effect, the main reason I believe throwmode was used is because forced you to pick between being offensive with it off or defensive with it on.

What if sleeping carp made you a light sleeper that wakes up faster than others (in combat mode only), sleeping carp users train themselves to sleep with their eyes open (like fish and thus carp do) and thus are more capable of being alert and ready at any times

**Comment by saprasam** [comment]
definitely makes sleeping carp better in a crew vs crew environment, but you are still going to get your ass kicked by a single dude with a baton, as nothing here really acts in a way comparable (or a counter to) how a baton is a 1shot down 2shot stun

**Comment by Redrover1760** [comment]
> definitely makes sleeping carp better in a crew vs crew environment, but you are still going to get your ass kicked by a single dude with a baton, as nothing here really acts in a way comparable (or a counter to) how a baton is a 1shot down 2shot stun

Well, 3 disarms will para someone and you can combo into a kill easily off any para, and the fact that disarm also has potential for wallshoves and stuff really makes keelhaul kind of terrifyingly powerful. At least on the aggressive.

**Comment by Yttriums** [comment]
hey i didn't read your pr because you've chosen to write it in an insufferable tone

**Pull request merged**